spec/enumerable_spec.rb

```diff
@@ -0,0 +1,168 @@
+require '../enumerables.rb'
+
+describe Enumerable do
+  test_array1 = [11, 2, 3, 56]
+  describe '#my_each' do
+    it 'returns the input array' do
+      expect(test_array1.my_each { |x| print x }).to eql(test_array1)
+    end
+  end
+
+  describe '#my_each_with_index' do
+    it 'returns the array and it index' do
+      expect(test_array1.my_each_with_index { |x, y| }).to eql(test_array1)
+    end
+  end
+
+  describe '#my_select' do
+    it 'returns filtered array based on supplied block' do
+      expect((0...10).my_select { |i| (i % 3) == 0 }).to eql([0, 3, 6, 9])
+    end
+  end
+
+  describe '#my_all' do
+    it 'returns true if all array items satisfy condition in block' do
+      expect(%w[ant bear cat].my_all? { |word| word.length >= 3 }).to eql(true)
+    end
+    it 'returns false if some of the array items do not satisfy condition in block' do
+      expect(%w[ant bear cat].my_all? { |word| word.length >= 4 }).to eql(false)
+    end
+    it 'returns true if all array items match regex argument' do
+      expect(%w[ant bear cat].my_all?(/a/)).to eql(true)
+    end
+    it "returns false if some array items don't match regex argument" do
+      expect(%w[ant bear cat].my_all?(/t/)).to eql(false)
+    end
+    it "returns false if all array items don't match regex argument" do
+      expect(%w[ant bear cat].my_all?(/z/)).to eql(false)
```

Comment:
- [ ] Instead of repeating the same example in all your tests, to follow DRY rule, you can declare a variable. Please use `let` to define them. [Here you can see more information about let](https://relishapp.com/rspec/rspec-core/v/2-10/docs/helper-methods/let-and-let)
![image](https://user-images.githubusercontent.com/53256139/91753596-bce10100-eb8d-11ea-81f0-f0438ad1ab3a.png)